### PR TITLE
Added login errors to login page, while keeping "next" url

### PR DESCRIPTION
**Pull request by lenagrech**

### CODE REVIEW

File: lms/lmsweb/views.py [hunk 1]
```diff
@@ -66,22 +66,35 @@ def load_user(user_id):
 
 
 @webapp.route('/login', methods=['GET', 'POST'])
-def login():
+def login(login_error=None):
```
Comment (yammesicka) [inline lms/lmsweb/views.py:69]: Please add typing to `login_error`

File: lms/lmsweb/views.py [hunk 2]
```diff
@@ -66,22 +66,35 @@ def load_user(user_id):
...
+            login_user(user)
+            return get_next_url(next_page)
+        elif user is not None and user.is_password_valid(password) is False:
+            login_error = 'Wrong password'
+            return redirect(
+                url_for('login', **{'next': next_page,
+                        'login_error': login_error}),
+            )
+        elif user is None:
+            login_error = 'Username does not exist, please register first'
```
Comment (yammesicka) [inline lms/lmsweb/views.py:91]: We shouldn't have a different message for user that doesn't exist, that's might lead to [username enumeration](https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/03-Identity_Management_Testing/04-Testing_for_Account_Enumeration_and_Guessable_User_Account).

File: lms/lmsweb/views.py [hunk 3]
```diff
@@ -66,22 +66,35 @@ def load_user(user_id):
...
-
-    return render_template('login.html')
+    if request.method == 'POST':
+        if user is not None and user.is_password_valid(password):
+            login_user(user)
+            return get_next_url(next_page)
+        elif user is not None and user.is_password_valid(password) is False:
+            login_error = 'Wrong password'
+            return redirect(
+                url_for('login', **{'next': next_page,
```
Comment (yammesicka) [inline lms/lmsweb/views.py:87]: Prefer:

```python
    url_for(
        'login', **{'next': next_page, 'login_error': login_error}),
    )
```

File: lms/templates/login.html [hunk 4]
```diff
@@ -27,6 +27,13 @@ <h1 id="main-title" class="h3 font-weight-normal">{{ _('התחברות') }}</h1>
           <input class="form-control form-control-lg" type="hidden" name="next" id="next" value="{{ request.args.get('next', '') }}">
           <button class="btn btn-primary btn-lg btn-block">{{ _('התחבר') }}</button>
         </form>
+        {% if login_error %}
```
Comment (yammesicka) [inline lms/templates/login.html:30]: Looking at different [error messages](https://www.google.com/search?q=login+error&tbm=isch) across the web, it's usually just below the login title or near the faulty field. I think we should follow this pattern.

**Review by yammesicka — CHANGES_REQUESTED** [review]

**Comment by yammesicka** [comment]
Great work :) Now to the enhancing phrase :man_dancing:

File: lms/lmsweb/views.py (hunk 2 above)
Comment (lenagrech) [reply lms/lmsweb/views.py:91]: wasn't aware of this kind of hacking, was an interesting read! changed the error message

File: lms/templates/login.html (hunk 4 above)
Comment (lenagrech) [reply lms/templates/login.html:30]: good point, I put it under the login

**Comment by yammesicka** [comment]
The build is currently failing, make it work before sending it again to review :)

**Comment by lenagrech** [comment]
I had an issue with my git work flow, i confused my remote and origin and wasn't therefore seeing any new updated on master.

i'll open anew merge request now that i fixed my repositories

**Pull request closed without merging**